### DISCUSSION TITLE
Issue in report creation

### DISCUSSION THREAD

**Discussion by agauttam** [utmstack/UTMStack#695] [Q&A] [answered]
Getting the below error on webpdf container

08:47:04.473 INFO [LocalDistributor.newSession] - Session request received by the Distributor:
 [Capabilities {acceptInsecureCerts: true, browserName: chrome, goog:chromeOptions: {args: [--headless, --no-sandbox], extensions: []}}]
08:47:06.104 INFO [LocalNode.newSession] - Session created by the Node. Id: 06580489d39ed5cb9422fd8fd8846d7b, Caps: Capabilities {acceptInsecureCerts: true, browserName: chrome-headless-shell, browserVersion: 124.0.6367.118, chrome: {chromedriverVersion: 124.0.6367.91 (51df0e5e17a8..., userDataDir: /tmp/.org.chromium.Chromium...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:42901}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: linux, proxy: Proxy(), se:bidiEnabled: false, se:cdp: ws://10.0.1.16:4444/session..., se:cdpVersion: 124.0.6367.118, se:vnc: ws://10.0.1.16:4444/session..., se:vncEnabled: true, se:vncLocalAddress: ws://10.0.1.16:7900, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
08:47:06.157 INFO [LocalDistributor.newSession] - Session created by the Distributor. Id: 06580489d39ed5cb9422fd8fd8846d7b
 Caps: Capabilities {acceptInsecureCerts: true, browserName: chrome-headless-shell, browserVersion: 124.0.6367.118, chrome: {chromedriverVersion: 124.0.6367.91 (51df0e5e17a8..., userDataDir: /tmp/.org.chromium.Chromium...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:42901}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: linux, proxy: Proxy(), se:bidiEnabled: false, se:cdp: ws://10.0.1.16:4444/session..., se:cdpVersion: 124.0.6367.118, se:vnc: ws://10.0.1.16:4444/session..., se:vncEnabled: true, se:vncLocalAddress: ws://10.0.1.16:7900, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
2024-05-29 08:47:06.459  INFO 8 --- [ null to remote] o.o.selenium.remote.ProtocolHandshake    : Detected dialect: W3C
08:48:05.058 WARN [SeleniumSpanExporter$1.lambda$export$3] - {"traceId": "32d93188bb5dd46556e7be8520aefadc","eventTime": 1716972484901432166,"eventName": "HTTP request execution complete","attributes": {"http.flavor": 1,"http.handler_class": "org.openqa.selenium.remote.http.Route$PredicatedRoute","http.host": "localhost:4444","http.method": "POST","http.request_content_length": "340","http.scheme": "HTTP","http.status_code": 500,"http.target": "\u002fsession\u002f06580489d39ed5cb9422fd8fd8846d7b\u002fprint","http.user_agent": "selenium\u002f4.1.4 (java unix)"}}

2024-05-29 08:48:07.575 ERROR 8 --- [nio-8080-exec-7] c.u.w.service.PdfGenerationService       : unknown error: session deleted because of page crash
from unknown error: cannot determine loading status
from tab crashed
  (Session info: chrome-headless-shell=124.0.6367.118)
Build info: version: '4.1.4', revision: '535d840ee2'
System info: host: '5bd53e6fb2db', ip: '10.0.1.16', os.name: 'Linux', os.arch: 'amd64', os.version: '5.15.0-107-generic', java.version: '17.0.11'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
Command: [06580489d39ed5cb9422fd8fd8846d7b, printPage {scale=1.0, orientation=portrait, margin=org.openqa.selenium.print.PageMargin@106069b2, page=org.openqa.selenium.print.PageSize@36d41067, shrinkToFit=true, background=false}]
Capabilities {acceptInsecureCerts: true, browserName: chrome-headless-shell, browserVersion: 124.0.6367.118, chrome: {chromedriverVersion: 124.0.6367.91 (51df0e5e17a8..., userDataDir: /tmp/.org.chromium.Chromium...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:42901}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, proxy: Proxy(), se:bidiEnabled: false, se:cdp: ws://10.0.1.16:4444/session..., se:cdpVersion: 124.0.6367.118, se:vnc: ws://10.0.1.16:4444/session..., se:vncEnabled: true, se:vncLocalAddress: ws://10.0.1.16:7900, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 06580489d39ed5cb9422fd8fd8846d7b
08:48:08.836 INFO [LocalSessionMap.lambda$new$0] - Deleted session from local Session Map, Id: 06580489d39ed5cb9422fd8fd8846d7b
08:48:08.866 INFO [GridModel.release] - Releasing slot for session id 06580489d39ed5cb9422fd8fd8846d7b
08:48:08.876 INFO [SessionSlot.stop] - Stopping session 06580489d39ed5cb9422fd8fd8846d7b

**Comment by c3s4rfred** [comment]
It looks that a network issue is avoiding the communication with the selenium, please, check that the vlan created during installation is working. Best regards.

**Comment by c3s4rfred** [comment]
Also, it is a common issue caused when selenium don't have enough shared memory, please use the latest version of the installer, that issue was fixed. Check the the compose.yml generated by your installer, in the web-pdf service you must see a volume pointing to /dev/shm, if not, install the latest version.

**Reply by agauttam** [reply]
Hi Freddy,

Its a random issue and the compose.yml is pointing the to /dev/shm path.
Few reports i am able to fetch but not all.
/utmstack/tmpfs:/dev/shm

Regards,
Arun Gauttam

**Reply by c3s4rfred** [reply]
Please tell us your system properties and the UTMStack version you are using.

**Reply by agauttam** [reply]
Hi Freedy,

UTM Ver 10.4.2-2024050917594
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy
Container detail: utmstack_web-pdf , RAM- 2GiB 
Hardware: Windows Hypervisor, CPU-16 core, RAM-128 GB, HDD 1.4 TB.
I have checked with both default ubuntu 22.04.4 kernel 5.15.0-107-generic and Microsoft hypervisor supported kernel 6.5.0-1022-azure.


Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy
root@sectm:/home/opmanager# dmidecode -t2
# dmidecode 3.3
Getting SMBIOS data from sysfs.
SMBIOS 3.1.1 present.

Handle 0x0003, DMI type 2, 17 bytes
Base Board Information
        Manufacturer: Microsoft Corporation
        Product Name: Virtual Machine
        Version: Hyper-V UEFI Release v4.0
        Serial Number: XXX-XXXX-XXXX-XXXX-XX
        Asset Tag: None
        Features:
                Board is a hosting board
        Location In Chassis: Virtual Machine
        Chassis Handle: 0x0002
        Type: Motherboard
        Contained Object Handles: 0

Regards,
Arun

**Reply by c3s4rfred** [reply]
If you have enough memory, you can try to add more memory to the web-pdf swarm service, in the resource -> limits section, then restart the service and be sure that the change was applied in the compose.yml file.

**Reply by agauttam** [reply]
Hi Freedy,

I have tested with 4 GB memory as well and suspecting its only printing the small size reports, example: printing report for threat activity but not for windows activity, o365 exchange, sharepoint. 

Regards,
Arun

**Comment by c3s4rfred** [comment]
Hi @agauttam we are working to reproduce that behavior, in our environments with normal installation all reports are generated fine we will let you know, if you can, send us an screenshot of the dashboard that don't generates the pdf.

**Reply by agauttam** [reply]
Hi,

Please find the requested snapshots.
1. Utmstack overview report >> working
2. Threatmgmt report >> non-working


![UTM_Threat](https://github.com/utmstack/UTMStack/assets/168720506/d10ee3ea-300c-465d-b88d-f35be9e52697)
![UTM_overview](https://github.com/utmstack/UTMStack/assets/168720506/e65b77e7-4c96-4820-9b0e-6c5b2561c8a6)

**Comment by agauttam** [comment]
Hi Freddy,

I compared the logs from both the working and non-working scenarios and found them to be similar. I also reviewed the package but could not find any way to debug the web-pdf logs.

Best,
Arun

**Reply by c3s4rfred** [reply]
Hi @agauttam, for some reason in your environment the web-pdf docker image can't generate the pdf for some dashboards. The queries for the Theat Activity dashboard has a high resources costs, more than the Overview, that's why you can generate that one. It is a Selenium issue,  we are working on that, we will let you know if we find something.

**Reply by c3s4rfred** [reply]
Hi @agauttam we're making some improvements around this issue, will be available on version 11.0.0

Best regards

**Reply by c3s4rfred** [reply]
Hi @agauttam, I'm happy to tell you that we released a new version 10.5.6 with pdf export fixes, please upgrade and let us know if your issue is solved.

Best regards

**Reply by agauttam** [reply]
Hi Freddy,
I have upgraded to version 10.5.6. but still I am facing the issue.

Regards,
Arun

**Comment by c3s4rfred** [comment]
Hi @agauttam please, install the latest version (10.5.8) and let us know if the issue is solved.

Best regards

**Comment by c3s4rfred** [comment]
The export to pdf option was tested in DEV environment, version 10.5.12 and the same version in production, so the discussion will be close, please upgrade to latest version.

Best regards

**Comment by agauttam** [comment, marked as answer]
Hi Team,

Issue is resolved after upgrading to 10.7.2. It will be helpful if anyone can brief about any changes done for report printing.

it was not fixed till version 10.6.0

Regards,
Arun
